vault/core.go

```diff
@@ -947,19 +951,26 @@ func CreateCore(conf *CoreConfig) (*Core, error) {
 	if conf.NumRollbackWorkers == 0 {
 		conf.NumRollbackWorkers = RollbackDefaultNumWorkers
 	}
-	// Use imported logging deadlock if requested
-	var stateLock locking.RWMutex
-	if strings.Contains(conf.DetectDeadlocks, "statelock") {
-		stateLock = &locking.DeadlockRWMutex{}
-	} else {
-		stateLock = &locking.SyncRWMutex{}
-	}
 
 	effectiveSDKVersion := conf.EffectiveSDKVersion
 	if effectiveSDKVersion == "" {
 		effectiveSDKVersion = version.GetVersion().Version
 	}
 
+	var detectDeadlocks []string
+	if conf.DetectDeadlocks != "" {
+		conf.DetectDeadlocks = strings.Trim(conf.DetectDeadlocks, " ")
```

Comment:
Over-the-top-nitpick-that-you-shouldn't-at-all-feel-pressured-to-include-but-my-over-the-top-pedantry-compelled-me-to-include: `strings.TrimSpace()` might be a bit more thorough in this case.

Reply:
Done, thanks!